vital/util/tests/test_span.cxx

```diff
@@ -0,0 +1,114 @@
+
+// ----------------------------------------------------------------------------
+TEST ( span, construct )
+{
+  {
+    auto const test_span = span< double >{ 5.0, 7.0 };
+    EXPECT_EQ( 5.0, test_span.lower() );
+    EXPECT_EQ( 7.0, test_span.upper() ); }
+
+  {
+    auto const test_span = span< int >{ 9, 2 };
+    EXPECT_EQ( 2, test_span.lower() );
+    EXPECT_EQ( 9, test_span.upper() );
+    EXPECT_EQ( span< int >( 2, 9 ), test_span );
+  }
+
+  {
+    EXPECT_THROW( span< float >( fnan, 5.0f ), std::invalid_argument );
+    EXPECT_THROW( span< double >( 1.0, dnan ), std::invalid_argument );
+  }
+}
+
+// ----------------------------------------------------------------------------
+TEST ( span, truncate )
+{
+  {
+    auto test_span = span< double >{ 0.0, 15.0 };
+    test_span.truncate_lower( -1.0 );
+    EXPECT_EQ( span< double >( 0.0, 15.0 ), test_span );
+    test_span.truncate_lower( 0.0 );
+    EXPECT_EQ( span< double >( 0.0, 15.0 ), test_span );
+    test_span.truncate_lower( 5.0 );
+    EXPECT_EQ( span< double >( 5.0, 15.0 ), test_span );
+    test_span.truncate_lower( 15.0 );
+    EXPECT_EQ( span< double >( 15.0, 15.0 ), test_span );
+    EXPECT_THROW( test_span.truncate_lower( 16.0 ), std::invalid_argument );
+    EXPECT_THROW( test_span.truncate_lower( dnan ), std::invalid_argument ); }
+
+  {
+    auto test_span = span< double >{ 0.0, 15.0 };
+    test_span.truncate_upper( 20.0 );
+    EXPECT_EQ( span< double >( 0.0, 15.0 ), test_span );
+    test_span.truncate_upper( 15.0 );
+    EXPECT_EQ( span< double >( 0.0, 15.0 ), test_span );
+    test_span.truncate_upper( 5.0 );
+    EXPECT_EQ( span< double >( 0.0, 5.0 ), test_span );
+    test_span.truncate_upper( 0.0 );
+    EXPECT_EQ( span< double >( 0.0, 0.0 ), test_span );
+    EXPECT_THROW( test_span.truncate_upper( -1.0 ), std::invalid_argument );
+    EXPECT_THROW( test_span.truncate_upper( dnan ), std::invalid_argument );
+  }
+}
+
+// ----------------------------------------------------------------------------
+TEST ( span, encompass )
+{
+  auto test_span = span< double >{ 0.0, 15.0 };
+  test_span.encompass( 5.0 );
+  EXPECT_EQ( span< double >( 0.0, 15.0 ), test_span );
+  test_span.encompass( -100.0 );
+  EXPECT_EQ( span< double >( -100.0, 15.0 ), test_span );
+  test_span.encompass( 100.0 );
+  EXPECT_EQ( span< double >( -100.0, 100.0 ), test_span );
+  EXPECT_THROW( test_span.encompass( dnan ), std::invalid_argument );
+}
+
+// ----------------------------------------------------------------------------
+TEST ( span, contains )
+{
+  {
+    EXPECT_FALSE( span< int >( 0, 0 ).contains( 0 ) );
+    EXPECT_FALSE( span< int >( -1, 5 ).contains( 5 ) );
+    EXPECT_FALSE( span< int >( -1, 5 ).contains( 6 ) );
+    EXPECT_FALSE( span< int >( -1, 5 ).contains( -2 ) );
+    EXPECT_TRUE(  span< int >( -1, 5 ).contains( 4 ) );
+    EXPECT_TRUE(  span< int >( -1, 5 ).contains( -1 ) );
+    EXPECT_FALSE( span< double >( -dinf, dinf ).contains( dnan ) ); }
```

Comment:
```suggestion
    EXPECT_FALSE( span< double >( -dinf, dinf ).contains( dnan ) );
  }
```
...although these braces (both instances in this test case) don't seem necessary? (Unlike elsewhere, there don't appear to be any local variables to be scope-limited?)